faucet/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2019 Alain Brenzikofer
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use encointer_primitives::{
+	communities::CommunityIdentifier,
+	reputation_commitments::{DescriptorType, FromStr, PurposeIdType},
+	scheduler::CeremonyIndexType,
+};
+
+use core::marker::PhantomData;
+use frame_support::{
+	traits::{Currency, ExistenceRequirement::KeepAlive, Get},
+	PalletId,
+};
+use frame_system::{self as frame_system, ensure_signed};
+use log::info;
+use sp_runtime::{traits::AccountIdConversion, Saturating};
+use sp_std::convert::TryInto;
+
+// Logger target
+const LOG: &str = "encointer";
+
+pub use pallet::*;
+
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + encointer_reputation_commitments::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type Currency: Currency<Self::AccountId>;
+		type FaucetMaster: EnsureOrigin<Self::RuntimeOrigin>;
```

Comment:
I believe the polkadot convention is to have `Origin` in the naming of origins. We may want to align this accross all pallets to just say `ControllerOrigin` ?

Reply:
makes sense